lib/reform/contract/validate.rb

```diff
@@ -33,7 +33,7 @@ def validate!(name, pointers = [])
       nested_errors = validate_nested!(pointers_for_nested)
 
       # Result: unified interface #success?, #messages, etc.
-      @result = Result.new(local_errors_by_group + pointers, nested_errors)
+      @result = Result.new(@result.to_result + local_errors_by_group + pointers, nested_errors)
```

Comment:
this is the part where we add `CustomError` before `validate` is called - by default that will return `[]`